community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v3_0/TransactionBoundPlanContext.scala

```diff
@@ -127,8 +127,14 @@ class TransactionBoundPlanContext(initialStatement: Statement, val gdb: GraphDat
     val ks = _statement.readOperations().procedureGet(kn)
     val input = ks.inputSignature().asScala.map(s => FieldSignature(s.name(), asCypherType(s.neo4jType())))
     val output = ks.outputSignature().asScala.map(s => FieldSignature(s.name(), asCypherType(s.neo4jType())))
+    val mode = asCypherProcMode(ks.mode())
 
-    ProcedureSignature(name, input, output)
+    ProcedureSignature(name, input, output, mode)
+  }
+
+  private def asCypherProcMode( mode:KernelProcedureSignature.Mode ): ProcedureMode = mode match {
+    case KernelProcedureSignature.Mode.READ_ONLY => PROC_READ_ONLY
+    case KernelProcedureSignature.Mode.READ_WRITE => PROC_READ_WRITE
```

Comment:
This is an exhaustive match currently, but perhaps it's a good idea to add a default case anyway, for a future implementor of another mode?